feat: client-side multi-provider implementation (#942)
open-feature/js-sdk-contrib: libs/providers/multi-provider-web/README.md

# OpenFeature Multi-Provider

> The Multi-Provider allows you to use multiple underlying providers as sources of flag data for the OpenFeature server SDK.

Reviewer: ```suggestion
The Multi-Provider allows you to use multiple underlying providers as sources of flag data for the OpenFeature web SDK.
```

Author: good catch - fixed in latest commit amendment

When a flag is being evaluated, the Multi-Provider will consult each underlying provider it is managing in order to determine
the final result. Different evaluation strategies can be defined to control which providers get evaluated and which result is used.

The Multi-Provider is a powerful tool for performing migrations between flag providers, or combining multiple providers into a single
feature flagging interface. For example:
- *Migration*: When migrating between two providers, you can run both in parallel under a unified flagging interface. As flags are added to the
new provider, the Multi-Provider will automatically find and return them, falling back to the old provider if the new provider does not have
- *Multiple Data Sources*: The Multi-Provider allows you to seamlessly combine many sources of flagging data, such as environment variables, 
local files, database values and SaaS hosted feature management systems.

## Installation

```
$ npm install @openfeature/multi-provider-web
```

> [!TIP]

> > This provider is designed to be used with the [Node.js SDK](https://openfeature.dev/docs/reference/technologies/server/javascript/).

Reviewer: ```suggestion
> This provider is designed to be used with the [Web SDK](https://openfeature.dev/docs/reference/technologies/client/web/).
```

## Usage
The Multi-Provider is initialized with an array of providers it should evaluate:

```typescript
import { WebMultiProvider } from '@openfeature/multi-provider-web'
import { OpenFeature } from '@openfeature/web-sdk'

const multiProvider = new WebMultiProvider([
  {
      provider: new ProviderA()
  },
  {
      provider: new ProviderB()
  }
])

await OpenFeature.setProviderAndWait(multiProvider)

const client = OpenFeature.getClient()

console.log("Evaluating flag")

> console.log(await client.getBooleanDetails("my-flag", false));

Reviewer: ```suggestion
console.log(client.getBooleanDetails("my-flag", false));
```
